sdk/python/feast/infra/registry_stores/sql.py

```diff
@@ -0,0 +1,413 @@
+
+        return self._apply_object(
+            fv_table, "feature_view_name", feature_view, "feature_view_proto"
+        )
+
+    def apply_feature_service(
+        self, feature_service: FeatureService, project: str, commit: bool = True
+    ):
+        return self._apply_object(
+            feature_services,
+            "feature_service_name",
+            feature_service,
+            "feature_service_proto",
+        )
+
+    def delete_data_source(self, name: str, project: str, commit: bool = True):
+        with self.engine.connect() as conn:
+            stmt = delete(data_sources).where(data_sources.c.entity_id == name)
+            rows = conn.execute(stmt)
+            if rows.rowcount < 1:
+                raise DataSourceObjectNotFoundException(name, project)
+
+    def list_feature_services(
+        self, project: str, allow_cache: bool = False
+    ) -> List[FeatureService]:
+        return self._list_objects(
+            feature_services,
+            FeatureServiceProto,
+            FeatureService,
+            "feature_service_proto",
+        )
+
+    def list_feature_views(
+        self, project: str, allow_cache: bool = False
+    ) -> List[FeatureView]:
+        return self._list_objects(
+            feature_views, FeatureViewProto, FeatureView, "feature_view_proto"
+        )
+
+    def list_saved_datasets(
+        self, project: str, allow_cache: bool = False
+    ) -> List[SavedDataset]:
+        return self._list_objects(
+            saved_datasets, SavedDatasetProto, SavedDataset, "saved_dataset_proto"
+        )
+
+    def list_request_feature_views(
+        self, project: str, allow_cache: bool = False
+    ) -> List[RequestFeatureView]:
+        return self._list_objects(
+            request_feature_views,
+            RequestFeatureViewProto,
+            RequestFeatureView,
+            "feature_view_proto",
+        )
+
+    def list_on_demand_feature_views(
+        self, project: str, allow_cache: bool = False
+    ) -> List[OnDemandFeatureView]:
+        return self._list_objects(
+            on_demand_feature_views,
+            OnDemandFeatureViewProto,
+            OnDemandFeatureView,
+            "feature_view_proto",
+        )
+
+    def apply_saved_dataset(
+        self, saved_dataset: SavedDataset, project: str, commit: bool = True,
+    ):
+        return self._apply_object(
+            saved_datasets, "saved_dataset_name", saved_dataset, "saved_dataset_proto"
+        )
+
+    def apply_validation_reference(
+        self,
+        validation_reference: ValidationReference,
+        project: str,
+        commit: bool = True,
+    ):
+        return self._apply_object(
+            validation_references,
+            "validation_reference_name",
+            validation_reference,
+            "validation_reference_proto",
+        )
+
+    def _apply_object(
+        self, table, id_field_name, obj, proto_field_name,
+    ):
```

Comment:
would like a docstring here, am fine with omitting them for now for other methods since they already exist in `registry.py`

Reply:
This is an internal method that's only meant to reduce duplication, but I can add some comments

Reply:
ah yeah maybe add some comments in future PR, not super high priority